core/batcher.go

```diff
@@ -0,0 +1,348 @@
+package core
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/anoideaopen/foundation/core/cachestub"
+	"github.com/anoideaopen/foundation/core/telemetry"
+	"github.com/anoideaopen/foundation/core/types"
+	"github.com/anoideaopen/foundation/hlfcreator"
+	"github.com/anoideaopen/foundation/internal/config"
+	"github.com/anoideaopen/foundation/proto"
+	pb "github.com/golang/protobuf/proto" //nolint:staticcheck
```

Comment:
I would recommend putting these changes in a separate package. Can we use the latest version of the protobuf library?

Reply:
I can't put batcher to separate pkg because after that I catch circular dependency problems.
And in my task "batcher" not enough time to make refactoring everething needed.

By this reason I use the same code part like already used 69 in foundation library like:

```
pb "github.com/golang/protobuf/proto" //nolint:staticcheck
```

Reply:
The `batcher` clearly belongs in a separate package, following Go's guidelines for modularity and separation of concerns. However, encountering circular dependencies when moving `batcher` suggests a design issue. According to Go best practices, circular dependencies indicate that the modules are too tightly coupled, which complicates the codebase. To adhere to Go's style and best practices, we should refactor to eliminate these dependencies, ensuring a cleaner and more maintainable architecture.